src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs

```diff
@@ -625,29 +626,44 @@ private void ResizeColumnToFit(int columnToResize)
 			FolderSettings.ColumnsViewModel = ColumnsViewModel;
 		}
 
-		private double MeasureTextColumnEstimate(int columnIndex, int measureItemsCount, int maxItemLength)
+		private double MeasureColumnEstimate(int columnIndex, int measureItemsCount, int maxItemLength)
+		{
+			if (columnIndex == 10)
+				return maxItemLength;
+
+			if (columnIndex == 3)
+				return MeasureTagColumnEstimate(columnIndex);
+
+			return MeasureTextColumnEstimate(columnIndex, measureItemsCount, maxItemLength);
+		}
+
+		private double MeasureTagColumnEstimate(int columnIndex)
 		{
-			var tbs = DependencyObjectHelpers.FindChildren<TextBlock>(FileList.ItemsPanelRoot).Where(tb =>
+			var grids = DependencyObjectHelpers.FindChildren<Grid>(FileList.ItemsPanelRoot).Where(grid => IsCorrectColumn(grid, columnIndex)).ToList();
```

Comment:
Going to a new line as you did with the Linq request l.645 to l.649 below is the way to go to make long calls like this one clearer. Please do it here as well.